### ISSUE TITLE
Deprecate 'lifted'?

### ISSUE THREAD

**Issue by Lev135** [ekmett/lens#1039]
As the [docs](https://hackage.haskell.org/package/lens-5.2.2/docs/Control-Lens-Setter.html#v:lifted) says

> You sometimes have to use this rather than 'mapped' -- due to temporary insanity 'Functor' was not a superclass of 'Monad' until  GHC 7.10.

Maybe this times has gone?

**Comment by RyanGlScott** [comment]
Yes, I think it would be appropriate to deprecate `lifted`. PRs welcome.
